Sources/Afterpay/Wrappers/ObjcWrapper.swift

```diff
@@ -168,6 +170,370 @@ public final class ObjcWrapper: NSObject {
+
+        case .failed(reason: .error(error: let error)):
+          completion(.failed(reason: .error(error: error)))
+        }
+      }
+    )
+  }
+
+  // Cash App signing end
+
+  // Cash App validate order
+
+  @objc(APCashAppValidationResultSuccess)
+  public class CashAppValidationResultSuccess: CashAppValidationResult {
+    @objc public let validationData: CashAppValidationData
+
+    init(validationData: CashAppValidationData) {
+      self.validationData = validationData
+    }
+  }
+
+  @objc(APCashAppValidationData)
+  public class CashAppValidationData: NSObject {
+    @objc public let cashAppTag: String
+    @objc public let status: String
+    @objc public let callbackBaseUrl: String
+
+    init(
+      cashAppTag: String,
+      status: String,
+      callbackBaseUrl: String
+    ) {
+      self.cashAppTag = cashAppTag
+      self.status = status
+      self.callbackBaseUrl = callbackBaseUrl
+    }
+  }
+
+  @objc(APCashAppValidationResult)
+  public class CashAppValidationResult: NSObject {
+    internal override init() {}
 
+    static func success(data: CashAppValidationData) -> CashAppValidationResultSuccess {
+      CashAppValidationResultSuccess(validationData: data)
+    }
+
+    static func failed(reason: CashAppValidationFailedReason) -> CashAppValidationResultFailed {
+      CashAppValidationResultFailed(reason: reason)
+    }
+  }
+
+  @objc(APCashAppValidationResultFailed)
+  public class CashAppValidationResultFailed: CashAppValidationResult {
+    @objc public let reason: CashAppValidationFailedReason
+
+    init(reason: CashAppValidationFailedReason) {
+      self.reason = reason
+    }
+  }
+
+  @objc(APCashAppValidationFailedReason)
+  public class CashAppValidationFailedReason: NSObject {
+    /// Should not be instantiated, instances should be of type CancellationReasonUserInitiated,
+    /// CancellationReasonNetworkError or CancellationReasonInvalidURL
+    internal override init() {}
+
+    static func nilData() -> CashAppValidationFailedReasonNilData {
+      CashAppValidationFailedReasonNilData(())
+    }
+
+    static func responseDecodeError() -> CashAppValidationFailedReasonResponseDecodeError {
+      CashAppValidationFailedReasonResponseDecodeError(())
+    }
+
+    static func unknownError() -> CashAppValidationFailedReasonUnknownError {
+      CashAppValidationFailedReasonUnknownError(())
+    }
+
+    static func invalid() -> CashAppValidationFailedReasonInvalid {
+      CashAppValidationFailedReasonInvalid(())
+    }
+
+    static func httpError(errorCode: Int, message: String) -> CashAppValidationFailedReasonHttpError {
+      CashAppValidationFailedReasonHttpError(errorCode, message)
+    }
+
+    static func error(error: Error) -> CashAppValidationFailedReasonError {
+      CashAppValidationFailedReasonError(error)
+    }
+  }
+
+  @objc(APCashAppValidationFailedReasonNilData)
+  public class CashAppValidationFailedReasonNilData: CashAppValidationFailedReason {
+    init(_ void: ()) {}
```

Comment:
Why do these take a void?

Reply:
no reason other than to keep consistency with the existing obj-c wrapper code.